src/core/Akka/Actor/SupervisorStrategy.cs

```diff
@@ -193,10 +190,9 @@ public static SupervisorStrategy DefaultStrategy
     /// </summary>
     public class OneForOneStrategy : SupervisorStrategy
     {
-        private readonly int _maxNumberOfRetries;
-        private readonly int _withinTimeRangeMilliseconds;
-        private readonly Func<Exception, Directive> _decider;
-        private readonly bool _loggingEnabled;
+        public int MaxNumberOfRetries { get; private set; }
+        public int WithinTimeRangeMilliseconds { get; private set; }
+        public IDecider Decider { get; private set; }
 
```

Comment:
Using JsonConverters is another option, that will take care of embedded primitives and the problem with using the correct ctor.